07-regression1.Rmd

```diff
@@ -527,21 +568,33 @@ sacr_spec <- nearest_neighbor(weight_func = "rectangular", neighbors = tune()) %
 Next, we'll use 5-fold cross-validation to choose the number of neighbours via the minimum RMSPE:
 ```{r 07-mult-cv}
 gridvals <- tibble(neighbors = seq(1, 200))
-sacr_k <- workflow() %>%
+sacr_multi <- workflow() %>%
   add_recipe(sacr_recipe) %>%
   add_model(sacr_spec) %>%
   tune_grid(sacr_vfold, grid = gridvals) %>%
   collect_metrics() %>%
   filter(.metric == "rmse") %>%
-  filter(mean == min(mean)) %>%
-  pull(neighbors)
-sacr_k
+  filter(mean == min(mean))
+sacr_k <- sacr_multi %>%
+              pull(neighbors)
+sacr_multi
 ```
-Here we see that the smallest RMSPE occurs when $K =$ `r sacr_k`.
+Here we see that the smallest estimated RMSPE from cross-validation occurs when $K =$ `r sacr_k`.
+If we want to compare this multivariate K-NN regression model to the model with only a single
+predictor *as part of the model tuning process* (e.g., if we are running forward selection as described
+in the chapter on evaluating and tuning classification models),
+then we must compare the accuracy estimated using only the training data via cross-validation.
+Looking back, the estimated cross-validation accuracy for the single-predictor 
+model was `r format(round(sacr_min$mean), big.mark=",", nsmall=0, scientific = FALSE)`.
+The estimated cross validation accuracy for the multivariate model is
+`r format(round(sacr_multi$mean), big.mark=",", nsmall=0, scientific = FALSE)`.
+Thus in this case, we did not improve the model 
+by a large amount by adding this additional predictor.
```

Comment:
Should we say why we are continuing with the analysis then - to illustrate that we can do multivariate predictions? And how we can do that?

Reply:
done